### DISCUSSION TITLE
Financial Address Mapper - Issues, Questions & Clarifications

### DISCUSSION THREAD

**Discussion by lalithkota** [Sunbird-RC/community#804] [General]
I have a couple of questions regarding Sunbird-RC's Financial Address Mapper, which is available [here](https://github.com/Sunbird-RC/g2p-mapper-registry).

- We have installed all the components described in the docker compose, and explored the available UI and Mapper APIs. We are able to call the link API and see the data being linked in the DB.
- We are unable to call the resolve API to get the resolution or the financial address. The yes_no response also doesn't seem to work. We are always getting a static account provider info output (because of [this](https://github.com/Sunbird-RC/g2p-mapper-registry/blob/366c34ff373776a052c716e0905bd08ffe1455eb/services/mapper-service/api/handlers/resolve.go#L88)(?))
- When using the asynchronous APIs described, the response always seems to be routed to a preconfigured callback service (because of [this](https://github.com/Sunbird-RC/g2p-mapper-registry/blob/366c34ff373776a052c716e0905bd08ffe1455eb/services/mapper-service/config/config.go#L9) (?)). It doesn't seem to be using the sender-uri/sender-id/receiver-id/url from the header (described in the G2P-Connect spec [here](https://github.com/G2P-Connect/specs/blob/3721fc4a9ad4ac21168e3e45b517ef6dae72b90f/release/yaml/mapper_core_api_v1.0.0.yaml#L1983) for example).
  - How do we give a different callback URL per each API call?
- The Signature value in the request (for any API call) seems to be ignored. The request seems to pass through, for any given value in the signature. Is this the desired output?
- Also as part of the latest draft of 1.0.0 of the Mapper API Spec, synchronous APIs are also added. For example [here](https://github.com/G2P-Connect/specs/blob/3721fc4a9ad4ac21168e3e45b517ef6dae72b90f/release/yaml/mapper_core_api_v1.0.0.yaml#L629). Are there any plans for Sunbird's impl to support this synchronous APIs?

How we can proceed with the above?

Thanks

**Comment by holashchand** [comment]
Hi @lalithkota 
1. Static account info is mocked.
2. The mapper service is created to use one provided service only. You will require to modify code to use different callback for given api.
3. The validation for signature is not done in this demo.

You can refer to this PR https://github.com/Sunbird-RC/community/issues/665 to know more about the functionalities implemented
